searchcore/src/vespa/searchcore/proton/matching/match_master.cpp

```diff
@@ -137,6 +139,10 @@ MatchMaster::getFeatureSet(const MatchToolsFactory &mtf,
             LOG(debug, "getFeatureSet: Did not find hit for docid '%u'. Skipping hit", docs[i]);
         }
     }
+    auto onSummaryTask = mtf.createOnSummaryTask();
+    if (onSummaryTask) {
```

Comment:
I would prefer to make onSummaryTask local to the if:
```
if (auto onSummaryTask = mtf.createOnSummaryTask()) {
    onSummaryTask->run(docs);
}
```

Reply:
also keep in mind that summary feature calculation will not happen during docsum fetching if there are no summary features to calculate...